### ISSUE TITLE
Datadog Chart [>3.10.9 <=3.16.0] : /var/log/datadog/cluster-agent.log read-only file system (RHOS)

### ISSUE THREAD

**Issue by jlamande** [DataDog/helm-charts#937]
**Describe what happened:**

Deployed datadog with helm chart 3.10.9. Cluster Agent is starting.

Deployed datadog with helm chart 3.16.0 with same values (helm configuration). Cluster agent fails starting with lot of logs like :

```
/var/log/datadog/cluster-agent.log read-only file system
```

Going back to chart 3.10.9 solves the issue.

**Additional environment details (Operating System, Cloud provider, etc):**

OpenShift 4.8 (Kubernetes 1.21)

**Comment by clamoriniere** [comment]
@jlamande 

thanks for the report we will investigate and fix is asap

**Comment by jlamande** [comment]
Hi @clamoriniere 

thanks for the quick fix 👍🏻 

Unfortunately, I have 2 other paths now listed as readonly : 

```
'/conf.d/kubernetes_state_core.yaml.default' -> '/etc/datadog-agent/conf.d/kubernetes_state_core.yaml.default'
cp: cannot create regular file '/etc/datadog-agent/conf.d/kubernetes_state_core.yaml.default': Read-only file system
...
log: exiting because of error: log: cannot create log: open /tmp/datadog-cluster-agent.datadog-cluster-agent-6c665dd7b8-vv2rg.1000690000.log.WARNING.20230309-034704.1: read-only file system
'/conf.d/kubernetes_state_core.yaml.default' -> '/etc/datadog-agent/conf.d/kubernetes_state_core.yaml.default'
```

**Comment by clamoriniere** [comment]
Hi @jlamande 

sorry about this issue. We will work on solving the issue. 

Could you confirm that CRIO is used as container runtime in your Openshift cluster?

**Comment by L3n41c** [comment]
Hello @jlamande !

Sorry for this annoying issue. It seems we’ve been hit by a `cri-o` specificity.
I hope to have fixed this issue thanks to #943.
Would you mind making a new test with the version `3.19.1` of the Helm chart ?
